### PR TITLE
Fix bucket_name bucketname consistency

### PR DESCRIPTION
**Pull request by yeelauren**
Separating out into smaller PRs  :smile_cat: 

Refactor for consistency see: https://github.com/coralnet/pyspacer/blob/cb0b427b03f60bb92e2e9bdf1ccbb64e5b2230c3/spacer/storage.py#L187

https://github.com/coralnet/pyspacer/blob/cb0b427b03f60bb92e2e9bdf1ccbb64e5b2230c3/spacer/storage.py#L205

https://github.com/coralnet/pyspacer/blob/cb0b427b03f60bb92e2e9bdf1ccbb64e5b2230c3/spacer/messages.py#L25

### CODE REVIEW

**Comment by yeelauren** [comment]
Additionally a new change from #85 ircc will also need to be changed  https://github.com/coralnet/pyspacer/blob/478a8f6f01d7bdc1cb7b40866d36d89b53dda314/spacer/messages.py#L71

**Comment by StephenChan** [comment]
Thanks Lauren! This'll make reviewing easier.

I'm definitely onboard with making this param name consistent, but:

1. It looks like there are still a decent amount of `bucket_name` usages remaining: for example in `messages.py`, `storage.py`, `tests/test_storage.py`, `tests/utils.py`, and `README.md`.
2. Pardon my pickiness, but unless there's another reason I've overlooked (e.g. consistency with boto), I think I'd actually favor `bucket_name` over `bucketname`. The style goes better with the `storage_type` param in `DataLocation.__init__()` and `storage_factory()`; and originally there were more usages of `bucket_name` than `bucketname` (around 50 vs. 10).

**Comment by yeelauren** [comment]
@StephenChan I agree that makes more sense - the latest push should address those changes !

File: spacer/tasks.py [hunk 1]
```diff
@@ -3,8 +3,7 @@
 """
 import time
 import traceback
-from logging import getLogger
-
```
Comment (StephenChan) [inline spacer/tasks.py:7]: This `getLogger` import is still needed. (This was something I worked with recently, so maybe this deletion was a merge/rebase slip-up?)

File: spacer/storage.py [hunk 2]
```diff
@@ -184,12 +184,14 @@ def clear_memory_storage():
...
 
-def storage_factory(storage_type: str, bucketname: str | None = None):
+def storage_factory(storage_type: str, bucket_name: str | None = None):
 
     assert storage_type in config.STORAGE_TYPES
 
     if storage_type == 's3':
-        return S3Storage(bucketname=bucketname)
+        if bucket_name is None:
+            raise ValueError("bucket_name must be a string for s3 storage")
```
Comment (StephenChan) [inline spacer/storage.py:193]: This check works for me since it's related and useful.

File: spacer/tasks.py [hunk 3]
```diff
@@ -28,8 +27,14 @@ def extract_features(msg: ExtractFeaturesMsg) -> ExtractFeaturesReturnMsg:
 
     check_extract_inputs(img, msg.rowcols, msg.image_loc.key)
 
+    check_rowcols(msg.rowcols, img)
```
Comment (StephenChan) [inline spacer/tasks.py:30]: `check_rowcols()` was replaced with `check_extract_inputs()` in PR #70. So don't need this line.

File: spacer/tasks.py [hunk 4]
```diff
@@ -28,8 +27,14 @@ def extract_features(msg: ExtractFeaturesMsg) -> ExtractFeaturesReturnMsg:
...
+    if not isinstance(msg.extractor, Callable):
+        raise TypeError("msg.extractor must be callable")
+    
     with config.log_entry_and_exit('actual extraction'):
         features, return_msg = msg.extractor(img, msg.rowcols)
+    if not hasattr(features, 'store'):
+        raise TypeError("features object must have a store method")
```
Comment (StephenChan) [inline spacer/tasks.py:37]: These two type checks (well, instance/attribute checks to be exact) aren't that related to the rest of the PR.

They're also not really needed in my opinion - I think in Python, type checks don't hurt individually but they can add up to clutter the codebase somewhat. I know Oscar had several such checks in the codebase here and there, but in general I wasn't really keen on adding more. Sometimes they're a good idea with recent API changes (like PR #62 which admittedly relates to the first check here), but I'd like to think that's more of a worry for apps with larger userbases than pyspacer's.

**Review by StephenChan — CHANGES_REQUESTED** [review]
Thanks Lauren, looks good regarding the param naming! I reviewed the other bits of the PR.

**Review by StephenChan — APPROVED** [review]
Thanks!

**Pull request merged**